internal/protoschema/plugin/pluginsourceinfo/pluginsourceinfo.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pluginsourceinfo
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/bufbuild/protoplugin"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protodesc"
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+const FileExtension = ".sourceinfo.binpb"
```

Comment:
Needs Godoc - please ensure all of your public types have Godoc before putting up PRs here, will not comment for rest of files